### ISSUE TITLE
Get tests running in the respective packages, independently

### ISSUE THREAD

**Issue by eddieajau** [joomla/joomla-framework#4]
See the Date package for a good example.
- `phpunit.xml.dist` needs to be copied from the Date package into each package and test the running of tests.
- `bootstrap.php needs to be copied from`Date\Tests` to the corresponding package you are working on.
- Each test file needs to be namepaced, for example, for the Date package, the namespace is `\Joomla\Date\Tests`.

Complete:
- [ ] Application
- [ ] Archive
- [ ] Cache
- [x] Controller
- [x] Crypt
- [ ] Data
- [ ] Database
- [x] Date
- [x] DI
- [x] Event
- [x] Facebook
- [ ] Filesystem
- [ ] Filter
- [ ] Form
- [ ] Github
- [ ] Google
- [ ] Http
- [x] Image
- [ ] Input
- [ ] Keychain
- [ ] Language
- [ ] Ldap
- [ ] Linkedin
- [ ] Log
- [ ] Model
- [ ] Oauth1
- [ ] Oauth2
- [ ] Profiler
- [ ] Registry
- [ ] Router
- [ ] Session
- [ ] String
- [ ] Twitter
- [x] Uri
- [x] Utilities
- [ ] View

**Comment by eddieajau** [comment]
Database done. I've put in a new Joomla/Test package. It was just getting a bit crazy to fit up TestReflection and also the mock builders. I'm assuming we can only include that if "dev" is set. Anyway, see what you think. ONly took 6 hours straight to patch it all up.

**Comment by eddieajau** [comment]
Utilities and Registry done.

**Comment by eddieajau** [comment]
Data package done.

**Comment by eddieajau** [comment]
String package done.

**Comment by dongilbert** [comment]
Working on Controller

**Comment by dongilbert** [comment]
ehh - maybe scratch that. Not sure what to do about it's dependency on JFactory. :(

So, until we refactor the code, it's kind of not possible to make these tests separate.

(and I thought I was picking an easy one)

**Comment by mbabker** [comment]
Attempting Model (maybe I got the easy one :wink: )

**Comment by mbabker** [comment]
Working Application

**Comment by mbabker** [comment]
Http's done.  I actually finished one... LOL

**Comment by eddieajau** [comment]
Whoot! Another one bits the dust.

**Comment by mbabker** [comment]
Uri done.

**Comment by dongilbert** [comment]
I'll be working on the following packages tonight:

Google
Image

**Comment by mbabker** [comment]
Where do we stand with this one?

**Comment by dongilbert** [comment]
We'll see after we tag beta3 and try to run `composer install; phpunit` in each directory. :)

**Comment by dongilbert** [comment]
IIRC, the only remaining conflicts were due to the inability of composer to resolve dependencies.

**Comment by wilsonge** [comment]
So much phpcs to be fixed tho :P

**Comment by dongilbert** [comment]
Jump on it bro.

**Comment by mbabker** [comment]
Perfect low hanging fruit for a student wanting to get their feet wet in
the PHP world :-P

**Comment by dongilbert** [comment]
Exactly @mbabker. It's not that I don't want to do it (I do, in fact, want to), but it's a great opportunity for those who want to to build some cred and experience. Fixing CS is how I got to most LOC touched on the FW.

**Comment by wilsonge** [comment]
haha. If I have time I'll get on it. Still getting screwed by my essay at
the moment that's due on Friday + a couple of job applications I need to
rewrite my CV for due next week. And my need to learn a load of electronics
I claimed I already knew about for my final year project so I got a good
project haha.

Although just saying I was at a careers fair today - where I was scoffed
for learning PHP because Java is the language of professionals apparently :P
